# HG changeset patch
# Parent 9f1d990b1d9e4c97885d63d5ff0c33e3cad80e27
# User Ben Kelly <ben@wanderview.com>
Cache can now auto-close streams written to disk.


diff --git a/dom/cache/FileUtils.cpp b/dom/cache/FileUtils.cpp
--- a/dom/cache/FileUtils.cpp
+++ b/dom/cache/FileUtils.cpp
@@ -193,24 +193,20 @@ FileUtils::BodyStartWriteStream(const Qu
     nsCOMPtr<nsIOutputStream> buffered;
     rv = NS_NewBufferedOutputStream(getter_AddRefs(buffered), fileStream, 4096);
     if (NS_WARN_IF(NS_FAILED(rv))) { return rv; }
 
     fileStream = buffered.forget();
     mode = NS_ASYNCCOPY_VIA_WRITESEGMENTS;
   }
 
-  // TODO: we should be able to auto-close now...
-
-  // Note, we cannot auto-close the source stream here because some of
-  // our source streams must be closed on the PBackground worker thread.
   rv = NS_AsyncCopy(aSource, fileStream, NS_GetCurrentThread(), mode,
                     4096, // chunk size
                     aCallback, aClosure,
-                    false, true, // close streams
+                    true, true, // close streams
                     aCopyContextOut);
   if (NS_WARN_IF(NS_FAILED(rv))) { return rv; }
 
   return rv;
 }
 
 // static
 void
